presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestRandomizeNullKeyInOuterJoin.java

```diff
@@ -119,6 +120,36 @@ public void testLeftJoinOnSameKey()
                 false);
     }
 
+    @Test
+    public void testLeftJoinOnSameKeyJoinAsRightSideInput()
```

Comment:
This test will fail without the changes in this PR, and succeed after making the changes.